lib/widgets/stream_colors.dart

```diff
@@ -0,0 +1,223 @@
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_color_models/flutter_color_models.dart';
+
+import '../api/model/model.dart';
+import 'color.dart';
+
+/// A provider for [StreamColorSwatch]es that computes and caches them.
+abstract class StreamColorSwatches {
```

Comment:
I think "provider", like "factory", is a word I'd like to avoid here in favor of unpacking what it means :slightly_smiling_face:

Working from your explanation at https://github.com/zulip/zulip-flutter/pull/746#discussion_r1645230417 , here's a version I think would be clear:
```suggestion
/// Computes a [StreamColorSwatch] for any given color, with caching.
abstract class StreamColorSwatches {
```

Perhaps better yet, here's a slightly different way of looking at it:

```suggestion
/// A lazily-computed map from a stream's base color to a
/// corresponding [StreamColorSwatch].
abstract class StreamColorSwatches {
```

Reply:
Cool, yeah; I like the second one.